### PR TITLE
Adapt the reporting to the rendering of dashboards

### LINKED ISSUE #6556: [Remove discover] Implement new PDF reports in dashboards
#  Description
We need to adapt the current PDF report features so it can generate reports from embeddable dashboards. This will allow us to progressively deprecate the legacy discover dependencies and deprecate AngularJS in the plugins.

## Implementation restrictions

- The PDF reports capability must not use any AngularJS related feature.

## Tasks

- [x] Adapt the existing PDF report feature to use embeddable dashboards
- [x] Adapt the generation of PDF reports
  - [x] modules
  - [x] groups
  - [x] agents inventory
  - [x] configuration

### PR DESCRIPTION
**Pull request by Desvelao**
### Description
This pull request adapts the reporting to the rendering of dashboards.
 
Changes:    
- Adapt the Generate report button to use the search context
  - Create state and reducers into Redux store
  - Create hook to send the search context from a dashboard
     to the Generate report button
  - Remove disabledReport property and use the search context instead
  - Replace the mechanim to take the visualizations of the screenshots from the DOM
  - Remove the dependency injection of the unused AngularJS service to
    generate the reports
  - Adapt the report of agent inventory to use the data source
- Add abilities to the toast message when a PDF report is generated:
  - Open the report
  - Go to the Reporting application
- Remove unused AngularJS service to generate the reports
- Fix an error creating the image from map visualizations
- Fix an error about malformed row related to summary alerts tables
- Fix the query on GitHub > Dashboard did not take into account the date range filter
- Fix missing Generate report button on Office 365 > Dashboard
- Add support to Office 365 for agents according to the documentation
- Fix missing Generate report button on Inventory data

> [!WARNING]  
> This pull request depends on the implementations of dashboards https://github.com/wazuh/wazuh-dashboard-plugins/issues/6477.

### Issues Resolved
#6556
#6457

### Evidence
Report toast
image

### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to the Dashboard tab of each Module and generate a report. The visualizations visible on the dashboard should be included into the report. Verify the results on the report is matching the search context of the browser. | :black_circle: | :black_circle: | :black_circle: |
| Go to the Server management > Endpoints Summary, select an agent, then go to Configuration and click on the Export PDF button. Verify the data included in the report it is the expected one. | :black_circle: | :black_circle: | :black_circle: |
| Go to the Server management > Endpoints Summary, select an agent, then go to Invenotyr data and click on the Generate report button. Verify the data included in the report it is the expected one. | :black_circle: | :black_circle: | :black_circle: |
| Go to the Server management > Endpoints Groups, select a group, and click on the Export PDF button. Verify the data included in the report it is the expected one. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Go to the Dashboard tab of each Module and generate a report. The visualizations visible on the dashboard should be included into the report. Verify the results on the report is matching the search context of the browser.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to the Server management > Endpoints Summary, select an agent, then go to Configuration and click on the Export PDF button. Verify the data included in the report it is the expected one.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to the Server management > Endpoints Summary, select an agent, then go to Invenotyr data and click on the Generate report button. Verify the data included in the report it is the expected one.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to the Server management > Endpoints Groups, select a group, and click on the Export PDF button. Verify the data included in the report it is the expected one.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by Desvelao** [comment]
# Changes

- Replace query.getOpenSearchQuery from global source to use the search context

**Comment by Desvelao** [comment]
# Problem

After updating the development branch and trying to generate the PDF report on Theat Hunting application that has already replaced the dashboard, I got errors. It seems the old dashboard is rendered and causes that take screenshots of visualizations that are not redered, and this causes a problem when building the PDF report on the backend side.

This issue https://github.com/wazuh/wazuh-dashboard-plugins/pull/6594 should remove the render of the old dashboard, and this problem should disappear.

**Comment by asteriscos** [comment]
| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to the Dashboard tab of each Module and generate a report. The visualizations visible on the dashboard should be included into the report. Verify the results on the report is matching the search context of the browser. | :black_circle: | :black_circle: | :black_circle: |
| Go to the Server management > Endpoints Summary, select an agent, then go to Configuration and click on the Export PDF button. Verify the data included in the report it is the expected one. | :black_circle: | :black_circle: | :black_circle: |
| Go to the Server management > Endpoints Summary, select an agent, then go to Invenotyr data and click on the Generate report button. Verify the data included in the report it is the expected one. | :black_circle: | :black_circle: | :black_circle: |
| Go to the Server management > Endpoints Groups, select a group, and click on the Export PDF button. Verify the data included in the report it is the expected one. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Go to the Dashboard tab of each Module and generate a report. The visualizations visible on the dashboard should be included into the report. Verify the results on the report is matching the search context of the browser.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to the Server management > Endpoints Summary, select an agent, then go to Configuration and click on the Export PDF button. Verify the data included in the report it is the expected one.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to the Server management > Endpoints Summary, select an agent, then go to Invenotyr data and click on the Generate report button. Verify the data included in the report it is the expected one.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to the Server management > Endpoints Groups, select a group, and click on the Export PDF button. Verify the data included in the report it is the expected one.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,7 +44,10 @@ All notable changes to the Wazuh app project will be documented in this file.
 ### Fixed
 
 - Fixed the scripted fields disappear when the fields of the events index pattern was refreshed [#6237](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6237)
+  <<<<<<< HEAD
```
Comment (asteriscos) [inline CHANGELOG.md:47]: There seem to be some unresolved conflicts in the changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -42,7 +44,10 @@ All notable changes to the Wazuh app project will be documented in this file.
...
+- # Fixed an error of malformed tabe row on the generation of PDF reports [#6558](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6558)
 - Fixed the sample alerts scripts to generate valid IP ranges and file hashes [#6667](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6667)
+  > > > > > > > 16d03fb8ab4633f3f9955bd295636ba6f85fa45c
```
Comment (asteriscos) [inline CHANGELOG.md:50]: There seem to be some unresolved conflicts in the changelog.

**Review by asteriscos — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (Desvelao) [reply CHANGELOG.md]: Fixed here https://github.com/wazuh/wazuh-dashboard-plugins/pull/6558/commits/02387c47148deba990a6bdf2219a0a2c4f0d0d18.

File: CHANGELOG.md (hunk 1 above)
Comment (Desvelao) [reply CHANGELOG.md]: Fixed here https://github.com/wazuh/wazuh-dashboard-plugins/pull/6558/commits/02387c47148deba990a6bdf2219a0a2c4f0d0d18.

**Comment by Desvelao** [comment]
I was reviewing the missing button to generate report on Inventory data and this seems was removed when we refactored the AngularJS template related to agents on https://github.com/wazuh/wazuh-dashboard-plugins/commit/aee026821f491a9ffe1de7ee0455304a2261a8b9#diff-f2709f07ddcbc36d539eeca4fa1d4c2ff3baa5dd1cf302633d6ae2270630c632L31-L34.

I will talk with @lucianogorza about the context of removing the component that rendered the button and other things.

**Comment by asteriscos** [comment]
> I was reviewing the missing button to generate report on Inventory data and this seems was removed when we refactored the AngularJS template related to agents on [aee0268#diff-f2709f07ddcbc36d539eeca4fa1d4c2ff3baa5dd1cf302633d6ae2270630c632L31-L34](https://github.com/wazuh/wazuh-dashboard-plugins/commit/aee026821f491a9ffe1de7ee0455304a2261a8b9#diff-f2709f07ddcbc36d539eeca4fa1d4c2ff3baa5dd1cf302633d6ae2270630c632L31-L34).
> 
> I will talk with @lucianogorza about the context of removing the component that rendered the button and other things.

We will fix this in the next stage of the release.

**Comment by asteriscos** [comment]
| Test | Chrome | Firefox | 
| --- |  --- | --- | 
| Go to the Dashboard tab of each Module and generate a report. The visualizations visible on the dashboard should be included into the report. Verify the results on the report is matching the search context of the browser. | :black_circle: | :black_circle: | 
| Go to the Server management > Endpoints Summary, select an agent, then go to Configuration and click on the Export PDF button. Verify the data included in the report it is the expected one. | :green_circle: | :green_circle: | 
| Go to the Server management > Endpoints Summary, select an agent, then go to Inventory data and click on the Generate report button. Verify the data included in the report it is the expected one. | :yellow_circle: | :yellow_circle: | 
| Go to the Server management > Endpoints Groups, select a group, and click on the Export PDF button. Verify the data included in the report it is the expected one. | :green_circle: | :green_circle: | 

**Details**
<details>
<summary>:black_circle: Go to the Dashboard tab of each Module and generate a report. The visualizations visible on the dashboard should be included into the report. Verify the results on the report is matching the search context of the browser.</summary>

Chrome - :black_circle:




Firefox - :black_circle:



</details>

<details>
<summary>:green_circle: Go to the Server management > Endpoints Summary, select an agent, then go to Configuration and click on the Export PDF button. Verify the data included in the report it is the expected one.</summary>

Chrome - :green_circle:

[wazuh-agent-configuration-002-1715857506.pdf](https://github.com/wazuh/wazuh-dashboard-plugins/files/15334105/wazuh-agent-configuration-002-1715857506.pdf)



Firefox - :green_circle:

[wazuh-agent-configuration-001-1715857578.pdf](https://github.com/wazuh/wazuh-dashboard-plugins/files/15334109/wazuh-agent-configuration-001-1715857578.pdf)


</details>

<details>
<summary>:yellow_circle: Go to the Server management > Endpoints Summary, select an agent, then go to Inventory data and click on the Generate report button. Verify the data included in the report it is the expected one.</summary>


The button the generate the report no longer exists. This will be fixed in the next stage of the release.




</details>

<details>
<summary>:green_circle: Go to the Server management > Endpoints Groups, select a group, and click on the Export PDF button. Verify the data included in the report it is the expected one.</summary>

Chrome - :green_circle:

image



Firefox - :green_circle:

image


</details>

**Comment by JuanGarriuz** [comment]
### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to the Dashboard tab of each Module and generate a report. The visualizations visible on the dashboard should be included into the report. Verify the results on the report is matching the search context of the browser. | :black_circle: | :black_circle: | :black_circle: |
| Go to the Server management > Endpoints Summary, select an agent, then go to Configuration and click on the Export PDF button. Verify the data included in the report it is the expected one. | :green_circle: | :black_circle: | :black_circle: |
| Go to the Server management > Endpoints Summary, select an agent, then go to Invenotyr data and click on the Generate report button. Verify the data included in the report it is the expected one. | :green_circle: | :black_circle: | :black_circle: |
| Go to the Server management > Endpoints Groups, select a group, and click on the Export PDF button. Verify the data included in the report it is the expected one. | :green_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Go to the Dashboard tab of each Module and generate a report. The visualizations visible on the dashboard should be included into the report. Verify the results on the report is matching the search context of the browser.</summary>

Chrome - :black_circle:



Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to the Server management > Endpoints Summary, select an agent, then go to Configuration and click on the Export PDF button. Verify the data included in the report it is the expected one.</summary>

Chrome - :green_circle:

image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to the Server management > Endpoints Summary, select an agent, then go to Invenotyr data and click on the Generate report button. Verify the data included in the report it is the expected one.</summary>

Chrome - :green_circle:

image

image


Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Go to the Server management > Endpoints Groups, select a group, and click on the Export PDF button. Verify the data included in the report it is the expected one.</summary>

Chrome - :green_circle:

image
image


Firefox - :black_circle:

Safari - :black_circle:

</details>

**Review by asteriscos — APPROVED** [review]

**Review by asteriscos — APPROVED** [review]

**Pull request merged**